API documentation.md

```diff
@@ -1,60 +1,169 @@
-# API Documentation
+# Kiruna eXplorer API Documentation
+
+TODO: example of json request and responses
+TODO: write what are the mandatory parameters
+
+# Index
+
+1. [Base URL](#base-url)
+2. Collections   
+    2.1. [Colletion `documents`](#collection-documents)  
+    2.2. [Collection `links`](#collection-links)
+    2.3. [Collection `sessions`](#collection-sessions)
+    2.4. [Collection `users`](#collection-users)
+    
+<br/>
 
 # Base URL
 
 The base URL for all API requests is:
 
-`http://localhost:3000/new-project`
+`http://localhost:3000/kiruna-eXplorer`
+
+<br/>
 
-# Collection `template`
+# Collection `documents`
 
-Collection description
+A collection of documents representing key nodes in the relocation process of Kiruna. Each document corresponds to agreements, conflicts, consultations, material effects, or other documents that directly influence the project. Documents are categorized by type, including informative, prescriptive, and original resources, and are connected through a network of direct and collateral consequences, projections, and updates.
 
-## GET /template/{id}
+## GET `/documents`
 
-Endpoint description
+Retrieve all documents.
 
 ### Query parameters
 
-- `filter1` (optional): filter 1 description;
-- `filter2` (optional): filter 2 description.
+TODO: filters
+
+## Response body
+
+```json
+[
+    {
+        "id": 1,
+        "title": "",
+        "description": "",
+        "coordinates": {
+            "latitude": "",
+            "longitude": ""
+        },
+        "links": [
+            {
+                "targetNodeId": 2,
+                "linkType": "DIRECT"
+            },
+            ...
+        ],
+        //other properties
+    },
+    ...
+]
+```
+
+### Errors
+
+This API can return the following error codes:
+
+- `401 Unauthorized`: You are unauthorized.
+- `500 Internal Server Error`: An unexpected error occurred on the server.
+
+## GET `/documents/{id}`
+
+Retrieve a specific document by its unique identifier.
+
+### Query parameters
+
+TODO:
 
 ### Response body
 
 ```json
 {
-    "attr1": "attr1",
-    "attr2": "attr2"
+    "id": 1,
+    "title": "",
+    "description": "",
+    "coordinates": {
+        "latitude": "",
+        "longitude": ""
+    },
+    "links": [
+        {
+            "targetNodeId": 2,
+            "linkType": "DIRECT"
+        },
+        ...
+    ],
+    //other properties
 }
 ```
 
 ### Errors
 
 This API can return the following error codes:
 
+- `401 Unauthorized`: You are unauthorized.
+- `404 Not Found`: The requested document was not found.
+- `500 Internal Server Error`: An unexpected error occurred on the server.
+
+## POST `/documents`
+
+Create a new document.
+
+### Request body
+
+```json
+{
+    "id": 1,
```

Comment:
the id should be assigned by the server. This field needs to be removed imo

Reply:
Copy-paste error